### ISSUE TITLE
AlertDialog A11Y: Swipe order is not as expected in VoiceOver on iOS

### ISSUE THREAD

**Issue by guylivneh** [flutter/flutter#78229]
Internal bug: b/182334959

Repro Steps:
Turn on Talkback.
Display a Material AlertDialog via Flutter on iOS with AlertDialog's scrollable property set to false.

Observe: VoiceOver swipe order is not right - both when showDialog's barrierDismissible is set to true and to false.

Expected: Initial VoiceOver focus should be on the title, then the body, then the buttons from left to right (or right to left for RTL). Z gesture should close the dialog if barrierDismissible is set to true.

**Comment by HansMuller** [comment]
CC @chunhtai

**Comment by chunhtai** [comment]
<details>
<summary>repro</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyWidget(),
    );
  }
}

class MyWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
          child: ElevatedButton(
            onPressed: (){
              showDialog(
                context: context,
                builder: (BuildContext context) {
                  return AlertDialog(
                    title: Text("My title"),
                    content: Text("This is my message."),
                    actions: [
                      TextButton(
                        child: Text("Cancel"),
                        onPressed:  () {Navigator.of(context).pop();},
                      ),
                    ],
                  );
                },
              );
            },
            child: Text('show dialog'),
          )
      ),
    );
  }
}
```

</details>

**Comment by chunhtai** [comment]
There are three separate issues
1. dialog content semantics gets merged into the dialog body, which causes the content to be placed before the title in semantics traverse order
2. The push route somehow generated one frame without the page content before it generate the entire frame. Although this feels like a bug in model route, but the accessibility bridge need to be able to handle this case
3. Z gesture is not wired up to the alert dialog (or probably all the dialogs in general)
